groupbytraceprocessor test flaky on windows

```
2021-06-29T22:05:02.571Z	DEBUG	groupbytraceprocessor/event.go:129	recording current state of the queue	{"num-events": 0}
--- FAIL: TestPeriodicMetrics (0.04s)
    event_test.go:522: 
        	Error Trace:	event_test.go:522
        	            				event_test.go:471
        	Error:      	Not equal: 
        	            	expected: int(0)
        	            	actual  : float64(1)
        	Test:       	TestPeriodicMetrics
```

See https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector-contrib/16095/workflows/d72e1b69-fc8f-4e9c-8c08-be39f75762eb/jobs/140089

--- a/processor/groupbytraceprocessor/event_test.go
+++ b/processor/groupbytraceprocessor/event_test.go
@@ -427,7 +427,7 @@ func TestPeriodicMetrics(t *testing.T) {
 
 	// ensure that we are starting with a clean state
 	view.Unregister(views...)
-	view.Register(views...)
+	assert.NoError(t, view.Register(views...))
 
 	// try to be nice with the next consumer (test)
 	defer view.Unregister(views...)
@@ -461,14 +461,16 @@ func TestPeriodicMetrics(t *testing.T) {
 	go em.periodicMetrics()
 
 	// ensure our gauge is showing 1 item in the queue
-	time.Sleep(10 * time.Millisecond) // wait enough time for the gauge to be updated
-	assertGauge(t, 1, mNumEventsInQueue)
+	assert.Eventually(t, func() bool {
+		return getGaugeValue(t, mNumEventsInQueue) == 1
+	}, 1*time.Second, 10*time.Millisecond)
 
 	wg.Done() // release all events
-	time.Sleep(5 * time.Millisecond)
 
 	// ensure our gauge is now showing no items in the queue
-	assertGauge(t, 0, mNumEventsInQueue)
+	assert.Eventually(t, func() bool {
+		return getGaugeValue(t, mNumEventsInQueue) == 0
+	}, 1*time.Second, 10*time.Millisecond)
 
 	// signal and wait for the recursive call to finish
 	em.shutdownLock.Lock()
@@ -513,13 +515,12 @@ func TestDoWithTimeout(t *testing.T) {
 	assert.WithinDuration(t, start, time.Now(), 20*time.Millisecond)
 }
 
-func assertGauge(t *testing.T, expected int, gauge *stats.Int64Measure) {
+func getGaugeValue(t *testing.T, gauge *stats.Int64Measure) float64 {
 	viewData, err := view.RetrieveData("processor/groupbytrace/" + gauge.Name())
 	require.NoError(t, err)
 	require.Len(t, viewData, 1) // we expect exactly one data point, the last value
 
-	sum := viewData[0].Data.(*view.LastValueData)
-	assert.EqualValues(t, expected, sum.Value)
+	return viewData[0].Data.(*view.LastValueData).Value
 }
 
 func assertGaugeNotCreated(t *testing.T, gauge *stats.Int64Measure) {

